### ISSUE TITLE
Transfer learning to CIFAR-10 dataset

### ISSUE THREAD

**Issue by VSehwag** [google-research/simclr#7]
Thanks Ting for open-sourcing the code and checkpoints. From the pre-trained model, I have been trying to reproduce the results on transfer learning to datasets mentioned in the paper. Given the challenges there, I wonder whether it will possible to provide some additional instructions on transfer learning, in particular to low-dimensional datasets. 

In my attempt, I found the details in the paper enough to reproduce results within 0.5-1% of reported accuracy on most transfer learning dataset and linear evaluation on ImageNet itself. However, on transfer learning to CIFAR-10, the best accuracy on my end was 75% (reported in the paper ~90%). I am using ResNet50-1x as the feature extractor. Unfortunately, both the repo and the paper are quite thin on the details when it comes to transfer learning to low dimensional datasets.  These are the design choices I used:

1. Since CIFAR-10 have only 32x32 pixel images, I used constant padding to make it 224x224. Did you follow a different strategy (like upsampling)?
2. Extract features from the pooling layer after the last residual block ('default'). I also experimented with the features from the other blocks. Best accuracy (75%) achieved from features from third resnet block.

Along these design choices, I have conducted a thorough hyper-parameter search but met no success. I wonder whether you employed different design choices to report results on transfer learning to CIFAR-10/CIFAR-100 dataset?

**Comment by simonster** [comment]
We process CIFAR-10 in exactly the same way as the other datasets. For linear evaluation, we bicubically resample from 32x32 to 224x224. For fine-tuning, at training time we use standard random crops (10-100% of the image) and flips, and at test time we bicubically resample to 256x256 and take a 224x224 center crop. We always use the average-pooled features.

**Comment by kaushal-py** [comment]
> Thanks Ting for open-sourcing the code and checkpoints. From the pre-trained model, I have been trying to reproduce the results on transfer learning to datasets mentioned in the paper. Given the challenges there, I wonder whether it will possible to provide some additional instructions on transfer learning, in particular to low-dimensional datasets.
> 
> In my attempt, I found the details in the paper enough to reproduce results within 0.5-1% of reported accuracy on most transfer learning dataset and linear evaluation on ImageNet itself. However, on transfer learning to CIFAR-10, the best accuracy on my end was 75% (reported in the paper ~90%). I am using ResNet50-1x as the feature extractor. Unfortunately, both the repo and the paper are quite thin on the details when it comes to transfer learning to low dimensional datasets. These are the design choices I used:
> 
> 1. Since CIFAR-10 have only 32x32 pixel images, I used constant padding to make it 224x224. Did you follow a different strategy (like upsampling)?
> 2. Extract features from the pooling layer after the last residual block ('default'). I also experimented with the features from the other blocks. Best accuracy (75%) achieved from features from third resnet block.
> 
> Along these design choices, I have conducted a thorough hyper-parameter search but met no success. I wonder whether you employed different design choices to report results on transfer learning to CIFAR-10/CIFAR-100 dataset?

Hi @VSehwag, were you able to reproduce the results? When I tried reproducing, the accuracy I am getting is ~80%. I tried various optimizers; Adam, SGD and L-BFGS. I did exhaustive tuning for learning rate and weight decay. 
@simonster I tried using the crop sizes that you suggested. Still no luck. Any suggestions?

**Comment by VSehwag** [comment]
Upsampling (not constant padding) to 224x224 worked for me. In general, most ImageNet scale models use the upsampled version of CIFAR10.

**Comment by deepankarvarma** [comment]
Can you share the notebook for Transfer learning on CIFAR-10 using Pre-trained Resnet50-1x (present in the research paper) ? @VSehwag
